util/dvsim/SimCfg.py

```diff
@@ -640,10 +600,32 @@ def _gen_results(self, run_results):
             self.results_summary["Name"] = self._get_results_page_link(
                 self.results_summary["Name"])
 
-        # Append failures for triage
-        if results.fail_msgs:
-            fail_msgs = "\n## List of Failures\n" + ''.join(results.fail_msgs)
-            results_str += fail_msgs
+        # Append bucketized failures for triage, sorted by descending number
+        # of failures.
+        if results.buckets:
+            self.errors_seen = True
+            by_tests = sorted(results.buckets.items(),
+                              key=lambda i: len(i[1]),
+                              reverse=True)
+            fail_msgs = ["\n## Failure Buckets", ""]
+            for bucket, tests in by_tests:
+                fail_msgs.append(f"*   ```{bucket}```:")
```

Comment:
nit: extra spaces after `*` IMO is not needed. ditto on 617 and 620 
Not sure if this will get rendered properly as a bulleted code-block. I will suggest converting the md to html and take a look at it to confirm. 
```suggestion
                fail_msgs.append(f"* ```{bucket}```:")
```

Reply:
The extra spaces are per the g3doc style guide: https://g3doc.corp.google.com/corp/g3doc/docs/reference/style.md?cl=head#lists.

Reply:
This is GitHub flavored MD, not our internal one. I guess it does not matter.

Reply:
This is GitHub flavored markdown, not what we use internally. Also perhaps best not to paste our internal sites in this public repo :). I guess it is ok as long as it renders properly.

Reply:
Ah, the reference to the internal doc should not have been added, sorry.